From 000fe63b6fc57b09828930cacbab20c2ee6e6d15 Mon Sep 17 00:00:00 2001
From: Mark Rousskov <mark.simulacrum@gmail.com>
Date: Fri, 11 Oct 2019 17:41:55 -0400
Subject: [PATCH] Remove painful test that is not pulling its weight

Research suggests that we are not properly testing this case anyway, and
even if we were, it is unlikely that we will regress here -- or, perhaps
more accurately, if we do, I am uncertain that we care too much. It
definitely seems like an edge case, and one that is particularly
unlikely to occur as time goes on.
---
 .../linker-output-non-utf8/Makefile           | 23 -------------------
 .../linker-output-non-utf8/exec.rs            |  6 -----
 .../linker-output-non-utf8/library.rs         | 10 --------
 3 files changed, 39 deletions(-)
 delete mode 100644 src/test/run-make-fulldeps/linker-output-non-utf8/Makefile
 delete mode 100644 src/test/run-make-fulldeps/linker-output-non-utf8/exec.rs
 delete mode 100644 src/test/run-make-fulldeps/linker-output-non-utf8/library.rs

diff --git a/src/test/run-make-fulldeps/linker-output-non-utf8/Makefile b/src/test/run-make-fulldeps/linker-output-non-utf8/Makefile
deleted file mode 100644
index b47ce17ec8baa..0000000000000
--- a/src/test/run-make-fulldeps/linker-output-non-utf8/Makefile
+++ /dev/null
@@ -1,23 +0,0 @@
--include ../tools.mk
-
-# Make sure we don't ICE if the linker prints a non-UTF-8 error message.
-
-# ignore-windows
-#
-# This does not work in its current form on windows, possibly due to
-# gcc bugs or something about valid Windows paths.  See issue #29151
-# for more information.
-
-# ignore-macos
-#
-# This also does not work on Apple APFS due to the filesystem requiring
-# valid UTF-8 paths.
-
-# The zzz it to allow humans to tab complete or glob this thing.
-bad_dir := $(TMPDIR)/zzz$$'\xff'
-
-all:
-	$(RUSTC) library.rs
-	mkdir $(bad_dir)
-	mv $(TMPDIR)/liblibrary.a $(bad_dir)
-	$(RUSTC) -L $(bad_dir) exec.rs 2>&1 | $(CGREP) this_symbol_not_defined
diff --git a/src/test/run-make-fulldeps/linker-output-non-utf8/exec.rs b/src/test/run-make-fulldeps/linker-output-non-utf8/exec.rs
deleted file mode 100644
index 6864018d64e97..0000000000000
--- a/src/test/run-make-fulldeps/linker-output-non-utf8/exec.rs
+++ /dev/null
@@ -1,6 +0,0 @@
-#[link(name="library")]
-extern "C" {
-    fn foo();
-}
-
-fn main() { unsafe { foo(); } }
diff --git a/src/test/run-make-fulldeps/linker-output-non-utf8/library.rs b/src/test/run-make-fulldeps/linker-output-non-utf8/library.rs
deleted file mode 100644
index 6689a82fa2c49..0000000000000
--- a/src/test/run-make-fulldeps/linker-output-non-utf8/library.rs
+++ /dev/null
@@ -1,10 +0,0 @@
-#![crate_type = "staticlib"]
-
-extern "C" {
-    fn this_symbol_not_defined();
-}
-
-#[no_mangle]
-pub extern "C" fn foo() {
-    unsafe { this_symbol_not_defined(); }
-}
